utils/frame/benchmarking-cli/src/storage/post_process.rs

```diff
@@ -0,0 +1,192 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Calculates statistics and fills out the `weight.hbs` template.
+
+use sc_cli::Result;
+use sc_service::Configuration;
+
+use clap::Args;
+use log::info;
+use serde::Serialize;
+use std::{collections::HashMap, fs, path::PathBuf};
+
+#[derive(Debug, Clone, PartialEq, Args)]
+pub struct PostProcParams {
+	/// Path to write the filled out weight template.
+	#[clap(long)]
+	pub weight_path: String,
+}
+
+/// Raw output of a Storage benchmark.
+#[derive(Debug, Default, Clone, Serialize)]
+pub struct TimeResult {
+	/// Multi-Map of value sizes and the time that it took to access them.
+	pub ns_by_size: Vec<(u64, u64)>,
+}
+
+#[derive(Serialize, Default, Debug, Clone)]
+pub struct TemplateData {
+	pub db_name: String,
+	pub runtime_name: String,
+	pub read: Option<TimeStats>,
+	pub write: Option<TimeStats>,
+}
+
+#[derive(Serialize, Default, Debug, Clone)]
+pub struct TimeStats {
+	pub min: u64,
+	pub max: u64,
+
+	pub avg: u64,
+	pub median: u64,
+	/// The mode is the most common value.
+	pub mode: u64,
```

Comment:
TBH mode is pretty strange with this data set

Reply:
Okay I can remove it, less code :laughing: